src/components/TransactionBlock/PlrDaoStakingTransactionBlock.tsx

```diff
@@ -368,19 +371,23 @@ const PlrDaoStakingTransactionBlock = ({
     [sdk, selectedFromAsset, amount, selectedFromNetwork, accountAddress, selectedAccountType],
   );
 
-  const getNftList = async () => {
-    try {
-      if (!accountAddress || !providerAddress || !sdk) return;
-      const output = await sdk.getNftList({
-        account: accountAddress || providerAddress,
+  const getNftList = () => {
+    if (!accountAddress || !providerAddress || !sdk) return Promise.resolve();
+
+    const nftPromiseArray = [providerAddress, accountAddress].map((address) =>
+      getNftsForChainId(CHAIN_ID.POLYGON, address, true)
+    );
+
+    Promise.all(nftPromiseArray)
+      .then(([providerNFTDetails, accountNFTDetails]) => {
+        const hasMembershipNFT = providerNFTDetails
+          .concat(accountNFTDetails)
+          .some((nft) => addressesEqual(nft.contractAddress, plrDaoMemberNFT.address));
+        setIsNFTMember(hasMembershipNFT);
+      })
+      .catch((error) => {
+        //
       });
```

Comment:
I'm sure this works fine, but personally I think writing it out this way makes it more clear + we don't need to have a separate promise array + we aren't wrapping the entire thing in a `Promise.all`
```suggestion
    try {
      const nftCollection = await Promise.all(
        [providerAddress, accountAddress].map(
          async (address) => await getNftsForChainId(CHAIN_ID.POLYGON, address, true)
        )
      ).then(([provider, account]) => provider.concat(account));

      setIsNFTMember(nftCollection.some((nft) => addressesEqual(nft.contractAddress, plrDaoMemberNFT.address)));
    } catch (e) {
      //
    }
```